doc/qos/ECN_and_WRED_statistics_HLD.md

```diff
@@ -0,0 +1,342 @@
+    - [Abbreviations](#abbreviations)
+    - [Overview](#overview)
+    - [Requirements](#requirements)
+    - [Architecture Design](#architecture-design)
+    - [High-Level Design](#high-level-design)
+      - [Changes in COUNTERS_DB](#changes-in-counters_db)
+      - [Changes in FLEX_COUNTER_DB](#changes-in-flex_counter_db)
+      - [Changes in STATE_DB](#changes-in-state_db)
+      - [Changes in Orchagent](#changes-in-orchagent)
+      - [CLI Changes](#cli-changes)
+        - [show queue counters on a ECN statistics supported platform](#show-queue-counters-on-a-ecn-statistics-supported-platform)
+        - [show queue counters on a platform which does not support ECN statistics](#show-queue-counters-on-a-platform-which-does-not-support-ecn-statistics)
+        - [show interface counters on a WRED drop statistics supported platform](#show-interface-counters-on-a-wred-drop-statistics-supported-platform)
+        - [show interface counters on a platform which does not support WRED drop statistics](#show-interface-counters-on-a-platform-which-does-not-support-wred-drop-statistics)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+
+### Revision
+
+| Rev | Date     | Author                      | Change Description |
+|:---:|:--------:|:---------------------------:|--------------------|
+| 0.1 |23/Feb/23 | Rajesh Perumal **(Marvell)**| Initial Version    |
+
+### Scope
+
+This document provides the high level design for ECN Queue Statistics, WRED Port Statistics in SONiC
+
+### Abbreviations
+
+
+
+|  Abbreviation | Description                     |
+|:-------------:|---------------------------------|
+| __ECN__       | Explicit Congestion Notification|
+| __WRED__      | Weighted Random Early Detection |
+| __CLI__       | Command Line interface          |
+
+
+### Overview
+
+The main goal of this feature is to provide better WRED impact visibility in SONiC by providing a mechanism to count the packets that are discarded or ECN-marked due to WRED.
+
+The other goal of this feature is to display these statistics only if the underlying platform supports it. [CLI changes](#cli-changes) section explains this in detail. Every platform may have unique statistics capabilities, and they change over time, and so it is important for this feature to be capability based.
+
+We will accomplish both the goals by adding support for per-queue ECN marked packets/bytes counters and per-port WRED dropped packets. Existing “queue counters” and “show interface counters detailed” CLI  will be enhanced for displaying these statistics.
+
+
+### Requirements
+
+1. Support per-queue total ECN marked packets counters

```

Comment:
Hi @rpmarvell, It will be better to emphasis that it counts the packets marked in the local device, since sometimes it was misunderstood as ECN bits equal to CE packets, which has happen in our env.

Reply:
I have updated the recent HLD with the suggested information. Please verify the same.